### PR TITLE
add support for yamllint

### PR DESCRIPTION
**Pull request by krzysztof-magosa**
This PR adds yaml-yamllint plugin (https://github.com/krzysztof-magosa/flycheck-yamllint/issues) to base flycheck code.

### CODE REVIEW

File: flycheck.el [hunk 1]
```diff
@@ -11238,6 +11239,22 @@ See URL `http://www.ruby-doc.org/stdlib-2.0.0/libdoc/yaml/rdoc/YAML.html'."
   :modes yaml-mode
   :next-checkers ((warning . cwl)))
 
+(flycheck-def-config-file-var flycheck-yamllintrc yaml-yamllint ".yamllint"
+  :safe #'stringp)
+
+(flycheck-define-checker yaml-yamllint
+  "A YAML syntax checker using YAMLLint.
+See URL `https://github.com/adrienverge/yamllint'."
+  :command (
```
Comment (cpitclaudel) [inline flycheck.el]: No need for a newline here

File: flycheck.el [hunk 2]
```diff
@@ -11238,6 +11239,22 @@ See URL `http://www.ruby-doc.org/stdlib-2.0.0/libdoc/yaml/rdoc/YAML.html'."
...
+            "yamllint" "-f" "parsable" source
+            (config-file "-c" flycheck-yamllintrc))
```
Comment (cpitclaudel) [inline flycheck.el]: From https://yamllint.readthedocs.io/en/stable/quickstart.html#running-yamllint it seems that yamllint supports checking standard input; could we use this?  This way we don't have to create temporary files.

File: flycheck.el [hunk 3]
```diff
@@ -11238,6 +11239,22 @@ See URL `http://www.ruby-doc.org/stdlib-2.0.0/libdoc/yaml/rdoc/YAML.html'."
...
+  :error-patterns
+  ((error line-start
+          (file-name) ":" line ":" column ": [error] " (message) line-end)
+   (warning line-start
+            (file-name) ":" line ":" column ": [warning] " (message) line-end))
```
Comment (cpitclaudel) [inline flycheck.el]: Does Yamllint support structured output, like JSON or XML? We tend to prefer those over plain text, when available.

File: flycheck.el (hunk 1 above)
Comment (krzysztof-magosa) [reply flycheck.el]: Otherwise it complains about line too long (80 characters).

File: flycheck.el (hunk 2 above)
Comment (krzysztof-magosa) [reply flycheck.el]: At the moment of writing of original code it didn't support stdin and author didn't have plan to implement it. I will update my PR to use stdin.

File: flycheck.el (hunk 3 above)
Comment (krzysztof-magosa) [reply flycheck.el]: It doesn't. `parsable` is mode supposed by author to be used for building linters etc.

**Comment by krzysztof-magosa** [comment]
@cpitclaudel I hope it looks correct now, please re-review.

File: flycheck.el (hunk 1 above)
Comment (cpitclaudel) [reply flycheck.el]: I meant between the `(` and `"yamllint"

File: flycheck.el (hunk 3 above)
Comment (cpitclaudel) [reply flycheck.el]: Got it, thanks!

**Comment by krzysztof-magosa** [comment]
I hope that's all ;)

**Comment by cpitclaudel** [comment]
Indeed it is! Thanks a lot :)

**Pull request merged**